File: java/user-management-policy/src/main/java/com/hexagonalchess/usermanagement/ListFriends.java
package com.hexagonalchess.usermanagement;

import java.util.List;

public class ListFriends {

    private final FriendshipRepository friendshipRepository;

    public ListFriends(FriendshipRepository friendshipRepository) {
        this.friendshipRepository = friendshipRepository;
    }

    public List<Friendship> execute(UserId ownerId) {
        return friendshipRepository.findByOwnerId(ownerId);
    }
}

File: python/user_management_policy/list_friends.py
from user_management_policy.friendship_repository import FriendshipRepository
from user_management_policy.user_id import UserId


def make_list_friends(friendship_repository: FriendshipRepository):
    def list_friends(owner_id: UserId) -> list:
        return friendship_repository.find_by_owner_id(owner_id)
    return list_friends
